src/yaml/from_yaml.cr

```diff
@@ -30,7 +30,8 @@ private def parse_yaml(string_or_io)
   end
 end
 
-private def parse_scalar(ctx, node, type : T.class) forall T
+private def parse_scalar(ctx, node, type : T.class,
+                         expected_type : Class? = nil) forall T
```

Comment:
Maybe this would be a bit more clearly express the function of the parameter and its default value?
```suggestion
                         expected_type : Class = T) forall T
```